packages/target-decisioning-engine/types/DecisioningConfig.d.ts

```diff
@@ -86,4 +87,9 @@ export interface DecisioningConfig {
    * Function used to parse domains
    */
   parseDomainImpl?: ParseDomainFunc;
+
+  /**
+   * Proxy Agent to specify a proxy for the fetch implementation
+   */
+  proxyAgent: ProxyAgent;
```

Comment:
Should probably also be nullable
```suggestion
  proxyAgent?: ProxyAgent;
```

Reply:
Do we need to add `undici` as a dependency and use this `ProxyAgent` object?  What is the benefit?  It is a standard?  I wonder if we can define a basic proxy of our own in target tools.

Reply:
[undici fetch](https://github.com/nodejs/undici#undicifetchinput-init-promise) is expecting an `undici` [Dispatcher](https://undici.nodejs.org/#/docs/api/Dispatcher) in `options.dispatcher`, which is the recommended solution for adding a proxy.

Reply:
`Dispatcher` extends node's `events.EventEmitter`, so it could be possible to omit the undici dependency through extending that, but that seems like it would be out of the scope of this ticket